validator/sawtooth_validator/networking/handlers.py

```diff
@@ -58,6 +61,27 @@ class ConnectHandler(Handler):
     def __init__(self, network):
         self._network = network
 
+    @staticmethod
+    def is_valid_endpoint_host(interfaces, endpoint):
+        """
+        An enpoint host name is valid if it is a URL and if the
+        host is not the name of a network interface.
+        """
+        result = urlparse(endpoint)
+        hostname = result.hostname
+        if hostname is None:
+            LOGGER.debug("Endpoint is not a valid URL : %s", endpoint)
```

Comment:
These `debug` statements should be `warning` or `error`, or else they might not be properly reported at validator startup.

Reply:
These messageslevels are fine, but there should be the appropriate error message at the point of usage of the function, not at such a low-level point.  The usage of this function has two wildly different effects: one is fatal, the other is merely a warning (and perfectly recoverable, as far as the caller is concerned)

Reply:
Yeah, leaving the logging at the call sites would be better.